openfed/openfed
# OPENFED-20: Fix encoding on file entity

### Fixes:

- Apply patch to fix wrong encoding when using Download link formatter plugin (see https://www.drupal.org/project/file_entity/issues/3277484)

### Improvements:

- Move file_entity related configurations to new submodule openfed_file_entity
- Remove file_entity dependency from openfed default configurations and openfed.info.yml
- Re-adapt hook openfed_update_101204()

### Test with a fresh installation of OpenFed 12:

- Fresh install OpenFed 12
- Make sure the file_entity module is not enabled by default
- Try to install openfed_file_entity and make sure the file_entity module is installed along with the same configurations that were moved from the openfed **_config/install_** folder into the new submodule configurations **_openfed_file_entity/config/optional_**
- Make sure the roles authenticated, anonymous have the "view files" permission by default
- Make sure the role content_editor has the "create files" permission by default
- Try to create a new media type with a file field and use the display formatter "Download link"
- Make sure you create a media field in one of your content types and allow the media type created previously
- Create a new page and upload a file with special characters in the filename in the media field
- When viewing the page, make sure the download link for the file is encoded correctly

### Test with an existing OpenFed site:

- Run `composer install`
- Run `drush updb`
- Make sure the hook openfed_update_101204() was successfully executed (test issue #39 to make sure it's working)
- Try to create a new media type with a file field and use the display formatter "Download link"
- Make sure you create a media field in one of your content types and allow the media type created previously
- Create a new page and upload a file with special characters in the filename in the media field
- When viewing the page, make sure the download link for the file is encoded correctly
